src/coll_patterns/recursive_knomial.h

```diff
@@ -230,6 +231,35 @@ ucc_knomial_calc_recv_dist(ucc_rank_t team_size, ucc_rank_t rank,
     return dist;
 }
 
+/* Calculates (sub) opt radix for Allreduce SRA and Bcast SAG,
+   by minimizing n_extra ranks */
+static inline ucc_rank_t ucc_kn_get_opt_radix(ucc_rank_t team_size,
+                                              ucc_kn_radix_t max_radix)
+{
+    ucc_rank_t     remainder = 0, min_val = 0;
+    ucc_kn_radix_t min_i     = UCC_KN_MIN_RADIX;
+    ucc_kn_radix_t r, fs;
+
+    max_radix = ucc_max(max_radix, UCC_KN_MIN_RADIX);
+
+    for (r = UCC_KN_MIN_RADIX; r <= max_radix; r++) {
+        fs = r;
+        while (fs < team_size) {
+            fs = fs * r;
+        }
+        fs        = (fs == team_size) ? fs : fs / r;
+        remainder = team_size - (team_size / fs) * fs;
+        if (remainder == 0) {
+            return r;
+        }
+        if (r == UCC_KN_MIN_RADIX || remainder < min_val) {
```

Comment:
i still don't see why not setting min_val = team_size at line 239, and then just keep this condition:
```
if (remainder < min_val) {
    min_val = remainder;
    min_i     = r;
}
```
and again, for me n_extra is better name than remainder, though i'm not insisting.

Reply:
You're correct, changed (also to n_extra :) )